packages/flutter/lib/src/material/navigation_rail_theme.dart

```diff
@@ -151,8 +151,10 @@ class NavigationRailThemeData with Diagnosticable {
       elevation: lerpDouble(a?.elevation, b?.elevation, t),
       unselectedLabelTextStyle: TextStyle.lerp(a?.unselectedLabelTextStyle, b?.unselectedLabelTextStyle, t),
       selectedLabelTextStyle: TextStyle.lerp(a?.selectedLabelTextStyle, b?.selectedLabelTextStyle, t),
-      unselectedIconTheme: IconThemeData.lerp(a?.unselectedIconTheme, b?.unselectedIconTheme, t),
-      selectedIconTheme: IconThemeData.lerp(a?.selectedIconTheme, b?.selectedIconTheme, t),
+      unselectedIconTheme: a?.unselectedIconTheme == null && b?.unselectedIconTheme == null
+        ? null : IconThemeData.lerp(a?.unselectedIconTheme, b?.unselectedIconTheme, t),
+      selectedIconTheme: a?.unselectedIconTheme == null && b?.unselectedIconTheme == null
```

Comment:
Should here be `a?.selectedIconTheme == null && b?.selectedIconTheme == null`?

Reply:
yes, good catch!